### PR TITLE
shitty print to pretty print

### PR DESCRIPTION
**Pull request by j-hui**
the long march

### CODE REVIEW

File: src/frontend/astprint.ml [hunk 1]
```diff
@@ -3,115 +3,149 @@ open Ast
...
-	| Neq -> "!="
-	| Leq -> "<="
-	| Geq -> ">="
-	| LogAnd -> "&&"
-	| LogOr -> "||"
-	| Filter -> "filter"
-	| Map -> "map"
-	| For -> "for"
-	| Do -> "do"
-	|	Asn -> "="
```
Comment (j-hui) [inline src/frontend/astprint.ml:45]: y i k e s

Comment (Contentmaudlin) [reply src/frontend/astprint.ml:45]: Yeah

File: src/frontend/parser.mly [hunk 2]
```diff
@@ -124,9 +124,9 @@ kn:
 
 lambda:
   | LPAREN formals RPAREN FUNC 
-    LBRACE statements ret_expr RBRACE       { LitKn({formals = $2; body = List.rev $6; ret_expr = $7}) } 
+    LBRACE statements ret_expr RBRACE       { LitKn({lformals = $2; lbody = List.rev $6; lret_expr = $7}) }
```
Comment (Contentmaudlin) [inline src/frontend/parser.mly:127]: Shouldn't you also update & commit `ast.mli` for this change?

Comment (Contentmaudlin) [reply src/frontend/parser.mly:127]: @j-hui

Comment (j-hui) [reply src/frontend/parser.mly:127]: already did

Comment (Contentmaudlin) [reply src/frontend/parser.mly:127]: OK. I don't see it but I trust you :D

**Review by Contentmaudlin — APPROVED** [review]
LGTM

**Comment by j-hui** [comment]
@Contentmaudlin i haven't actually tested this on any code yet – i don't know if it actually prints the proper stuff. but should i just merge now, and wait for it to crash and burn later?

File: src/frontend/ast.mli [hunk 3]
```diff
@@ -33,9 +33,9 @@ type un_op =
   | LogNot | Neg | Pos
 
 type lambda = {
-  formals   : bind list;
-  body      : stmt list;
-  ret_expr  : expr option;
+  lformals  : bind list;
+  lbody     : stmt list;
```
Comment (j-hui) [inline src/frontend/ast.mli:37]: @Contentmaudlin over hereee

Comment (Contentmaudlin) [reply src/frontend/ast.mli:37]: thx

**Review by Contentmaudlin — CHANGES_REQUESTED** [review]
bring back the wyt space pls

File: src/frontend/ast.mli [hunk 4]
```diff
@@ -82,15 +82,14 @@ type extern_decl = {
   exformals   : bind list;
 }
 
-
```
Comment (Contentmaudlin) [inline src/frontend/ast.mli:85]: Bring back the wyt space

**Review by Contentmaudlin — APPROVED** [review]

File: src/frontend/ast.mli (hunk 4 above)
Comment (Contentmaudlin) [reply src/frontend/ast.mli:85]: jk

Comment (j-hui) [reply src/frontend/ast.mli:85]: 👻

**Pull request merged**